File: Practice/Sources/Converter/app/src/main/java/com/example/converter/MainActivity.java
package com.example.converter;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
	public final static String UNITS_INTENT_EXTRA   = "com.example.converter.units";
	public final static String FACTORS_INTENT_EXTRA = "com.example.converter.factors";

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		setupButtonsEventHandlers();
	}

	private void setupButtonsEventHandlers() {
		Button lengthButton = findViewById(R.id.lengthButton);
		lengthButton.setOnClickListener(new View.OnClickListener() {

			@Override
			public void onClick(View view) {
				Intent intent = new Intent(view.getContext(), ConversionActivity.class);
				intent.putExtra(UNITS_INTENT_EXTRA,   R.array.length_conversion_unit_names);
				intent.putExtra(FACTORS_INTENT_EXTRA, R.array.length_conversion_factors);

				startActivity(intent);
			}

		});

		Button massButton = findViewById(R.id.massButton);
		massButton.setOnClickListener(new View.OnClickListener() {

			@Override
			public void onClick(View view) {
				Intent intent = new Intent(view.getContext(), ConversionActivity.class);
				intent.putExtra(UNITS_INTENT_EXTRA,   R.array.mass_conversion_unit_names);
				intent.putExtra(FACTORS_INTENT_EXTRA, R.array.length_conversion_factors);

				startActivity(intent);
			}

		});

		Button currencyButton = findViewById(R.id.currencyButton);
		currencyButton.setOnClickListener(new View.OnClickListener() {

			@Override
			public void onClick(View view) {
				Intent intent = new Intent(view.getContext(), CurrencyConversionActivity.class);
				startActivity(intent);
			}

		});
	}
}

File: Practice/Sources/Doodler/app/src/main/java/com/toksaitov/doodler/MainActivity.kt
package com.toksaitov.doodler

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.lifecycle.ViewModelProvider
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.toksaitov.doodler.data.Doodle
import com.toksaitov.doodler.data.DoodleViewModel
import java.io.File

class MainActivity : AppCompatActivity(), DoodleFragment.OnListFragmentInteractionListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportFragmentManager
            .beginTransaction()
            .add(R.id.placeholder, DoodleFragment.newInstance())
            .commit()

        val addDoodleButton : FloatingActionButton = findViewById(R.id.addDoodleButton)
        addDoodleButton.setOnClickListener {
            val intent = Intent(this, DoodleActivity::class.java)
            startActivity(intent);
        }
    }

    override fun onListFragmentClick(doodle: Doodle) {
        val intent = Intent(this, DoodleActivity::class.java).apply {
            putExtra("doodlePath", doodle.path)
        }
        startActivity(intent);
    }

    override fun onListFragmentSwipe(doodle: Doodle) {
        File(doodle.path).delete()
        ViewModelProvider(this).get(DoodleViewModel::class.java).delete(doodle)
    }

}
